### DISCUSSION TITLE
how to pass array of values using input from Text field?

### DISCUSSION THREAD

**Discussion by haizendaizen** [react-hook-form/react-hook-form#3160] [Q&A] [answered]
Hi All,

I've a working POST API endpoint that support the creation of data into a mongoDB instance by using the following request format:

```
{
   "name": "John",
   "description": "Test description",
   "uploads": [
       {
          "_id": "abc_file"
        }
    ]
 }
```

You'll notice "uploads" field is an array in the DB Schema.

To generate my POST request using a UI, I'm currently mapping the above fields into a Web form using useForm hook. 

OnSubmit function will call the mentioned API endpoint for creating the data.

I'm able to pass a single element into the "uploads" array by registering the form fields like this: 

<input type="text" name="**name**" ref={register({ required: true })} value="John" />
<input type="text" name="**description**" ref={register({ required: true })} value="Test description" />
<input type="text" name="**uploads._id**" ref={register({ required: true })} value="abc_file" />

That will work perfectly for a single array element and create the data in the DB without issues since it will create the proper POST request for the API.

I'm wondering if it's possible to pass multiple elements to the "uploads" array within my Web form?

Is there a way I can generate a POST request like this one with useForm?:

```
{
   "name": "John",
   "description": "Test description",
   "uploads": [
       {
          "_id": "abc_file"
        },
       {
          "_id": "def_file"
        }
    ]
 }
```

If i use multiple input fields registered with the same _id field I would only get the last one in my POST request:

```
<input type="text" name="**name**" ref={register({ required: true })} value="John" />
<input type="text" name="**description**" ref={register({ required: true })} value="Test description" />
<input type="text" name="**uploads._id**" ref={register({ required: true })} value="abc_file" />
<input type="text" name="**uploads._id**" ref={register({ required: true })} value="def_file" />
```
Will result in this POST request:

```
{
   "name": "John",
   "description": "Test description",
   "uploads": [
       {
          "_id": "def_file"
        }
    ]
 }
```

Any feedback or idea to implement is very welcome =)

Many thanks.

**Comment by bluebill1049** [comment]
`setValue('uplodates', { updates: [xxxx] })`

try above.

**Comment by haizendaizen** [comment]
Hey @bluebill1049 Thanks for pointing to this function. This could be the right path to solve this.

To simplify this example, the goal is to create the following POST request with the Web form:

```
{
   "uploads": [
       {
          "_id": "abc_file"
        },
       {
          "_id": "def_file"
        }
    ]
 }
```


So far I tried:

1) Single element in the uploads array. Note that I'm using dot notation for registering the field and get access to the nested field inside the array: https://codesandbox.io/s/quiet-darkness-92hcc?file=/src/App.js - **Works fine!**

`{"uploads":{"_id":"abc_file"}}`


2) Adding a second element to the uploads array with your suggestion: https://codesandbox.io/s/silly-faraday-tmql5?file=/src/App.js - **Array values are lost in the request and displaying as "object Object"**

`{"uploads":"[object Object]"}`

3) I also tried passing the values the way i need them with setValue: https://codesandbox.io/s/vibrant-williamson-dgpoq?file=/src/App.js - **Same Result as above**

`{"uploads":"[object Object]"}`

Any idea what could be wrong with my code?

Thanks again for your time.

**Comment by bluebill1049** [comment, marked as answer]
`uploads` is not a FieldArray.

**Reply by bluebill1049** [reply]
if you just want to store the value, i would just do a customer register at useEffect and store value with setValue, i will a codesandbox with it, if it doesn't work, then i am happy to patch that.

**Reply by bluebill1049** [reply]
custom register below:
https://codesandbox.io/s/inspiring-wood-4z0n0?file=/src/App.tsx

**Reply by haizendaizen** [reply]
This code illustrates exactly what I was looking for, thank you so much for your guidance!

**Reply by Tan-nine7soft** [reply]
Thank you very much. You are a true hero Bill.

**Reply by bardia1375** [reply]
perfect!
but If the uploads field is dynamic. . How to send its value to the server ؟
@bluebill1049 
for example user can add id
```
{
   "uploads": [
       {
          "_id": "use-add"
        },
       {
       "_id": "use-add"
        },
       {
       "_id": "use-add"
        },
       {
       "_id": "use-add"
        },
        ...
    ]
 }
```

**Comment by haizendaizen** [comment]
Solution for my specific scenario can be seen here: https://codesandbox.io/s/quiet-darkness-92hcc?file=/src/App.js
